File: tests/leap_compat/swift/ConversationConsumer.swift
import Foundation
import LeapSDK

final class ConversationConsumer: NSObject, Conversation {
  let delegate: any Conversation
  init(delegate: any Conversation) { self.delegate = delegate }
  var modelRunner: any ModelRunner { delegate.modelRunner }
  var generatingLock: any Kotlinx_coroutines_coreMutex { delegate.generatingLock }
  var history: [ChatMessage] { delegate.history }
  var functions: [LeapFunction] { delegate.functions }
  var isGenerating: Bool { delegate.isGenerating }

  func appendToHistory(message: ChatMessage) { delegate.appendToHistory(message: message) }
  func removeLastMessage() { delegate.removeLastMessage() }
  func registerFunction(function: LeapFunction) { delegate.registerFunction(function: function) }
  func registerFunctions(functions: [LeapFunction]) {
    delegate.registerFunctions(functions: functions)
  }

  func generateResponse(
    userTextMessage: String, generationOptions: GenerationOptions?
  ) -> SkieSwiftFlow<any MessageResponse> {
    delegate.generateResponse(
      userTextMessage: userTextMessage, generationOptions: generationOptions)
  }
  func generateResponse(
    message: ChatMessage, generationOptions: GenerationOptions?
  ) -> SkieSwiftFlow<any MessageResponse> {
    delegate.generateResponse(message: message, generationOptions: generationOptions)
  }
}

final class ParserConsumer: LeapFunctionCallParser {
  override func parse() -> [LeapFunctionCall] { [] }
  override func dump(functionCalls: [LeapFunctionCall]) -> String { "[]" }
}

File: tests/leap_compat/kotlin/ConversationConsumer.kt
package consumer

import ai.liquid.leap.Conversation
import ai.liquid.leap.GenerationOptions
import ai.liquid.leap.ModelRunner
import ai.liquid.leap.function.LeapFunction
import ai.liquid.leap.function.LeapFunctionCall
import ai.liquid.leap.function.LeapFunctionCallParser
import ai.liquid.leap.message.ChatMessage
import ai.liquid.leap.message.MessageResponse
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.sync.Mutex

class ConversationConsumer(
    private val delegate: Conversation,
) : Conversation {
    override val modelRunner: ModelRunner get() = delegate.modelRunner
    override val generatingLock: Mutex get() = delegate.generatingLock
    override val history: List<ChatMessage> get() = delegate.history
    override val functions: List<LeapFunction> get() = delegate.functions
    override val isGenerating: Boolean get() = delegate.isGenerating

    override fun appendToHistory(message: ChatMessage) = delegate.appendToHistory(message)

    override fun removeLastMessage() = delegate.removeLastMessage()

    override fun registerFunction(function: LeapFunction) = delegate.registerFunction(function)

    override fun registerFunctions(functions: List<LeapFunction>) = delegate.registerFunctions(functions)

    override fun generateResponse(
        userTextMessage: String,
        generationOptions: GenerationOptions?,
    ): Flow<MessageResponse> = delegate.generateResponse(userTextMessage, generationOptions)

    override fun generateResponse(
        message: ChatMessage,
        generationOptions: GenerationOptions?,
    ): Flow<MessageResponse> = delegate.generateResponse(message, generationOptions)
}

class ParserConsumer : LeapFunctionCallParser("<tool>", "</tool>") {
    override fun parse(): List<LeapFunctionCall> = emptyList()

    override fun dump(functionCalls: List<LeapFunctionCall>): String = "[]"
}
